src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -164,30 +222,55 @@
      * file.
      */
     public static final String MSG_SUMMARY_JAVADOC = "summary.javaDoc";
+
     /**
      * A key is pointing to the warning message text in "messages.properties"
      * file.
      */
     public static final String MSG_SUMMARY_JAVADOC_MISSING = "summary.javaDoc.missing";
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties" file.
+     */
+    public static final String MSG_SUMMARY_MISSING_PERIOD = "summary.javaDoc.missing.period";
+
     /**
      * This regexp is used to convert multiline javadoc to single line without stars.
      */
     private static final Pattern JAVADOC_MULTILINE_TO_SINGLELINE_PATTERN =
             Pattern.compile("\n[ ]+(\\*)|^[ ]+(\\*)");
 
+    /**
+     * This regexp is used to remove html tags, whitespace, and asterisks from a string.
+     */
+    private static final Pattern REDUNDANT_ELEMENTS_PATTERN =
+            Pattern.compile("<[^>]*>|(, + *)|(\\*)");
```

Comment:
content of summary like `,*` should not be removed.
Please place this as test case in Inputs.
Only leading asterisks are allowed to be skipped, not all asterisks.

rename REDUNDANT_ELEMENTS_PATTERN to NOT_HTML_PRINTABLE_ELEMENTS_PATTERN

Reply:
I am getting the entire summary by `getBlockComment()`. AST is not being used here. We can use the present regexp `JAVADOC_MULTILINE_TO_SINGLELINE_PATTERN` to remove asterisks, and new `HTML_ELEMENTS` to remove HTML tags.

Done.

Reply:
https://github.com/checkstyle/checkstyle/issues/9993